### PR TITLE
Add insertion sort using Laws

### PR DESCRIPTION
**Pull request by samuelchassot**
based on example in the introduction of Nicolas' thesis

### CODE REVIEW

File: algorithms/sorting/InsertionSort.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  def lemmaTest[T](x: T, y: T)(implicit ord: Ordering[T]): Unit = {
+    require(ord.compare(x, y) > 0)
+    assert(ord.inverse(x,y)) // If removed, a counter example is found
+  } ensuring (_ => ord.compare(y, x) < 0)
+
+  def isSorted[T](l: List[T])(implicit ord: Ordering[T]): Boolean = {
+    decreases(l)
+    l match
+    case Cons(x, Cons(y, ys)) => ord.compare(x, y) <= 0 && isSorted(Cons(y, ys))
+    case _                    => true
```
Comment (mario-bucev) [inline algorithms/sorting/InsertionSort.scala:33]: There is a missing indentation, I'm actually surprised that it compiles under Scala 3!

Comment (samuelchassot) [reply algorithms/sorting/InsertionSort.scala]: whoops indeed. In fact the scalafmt config is for scala 2, so it does not format scala 3.
Should we update?

Comment (mario-bucev) [reply algorithms/sorting/InsertionSort.scala]: I do not see any scalafmt file in this directory but if you wish you can add it :)

Comment (samuelchassot) [reply algorithms/sorting/InsertionSort.scala]: ah yes, it was only local. I commited :)

**Pull request merged**